### ISSUE TITLE
🔥 This Static and Gzip makes me desperate

### ISSUE THREAD

**Issue by hyingreborn** [gofiber/fiber#222]
This Static and Gzip makes me desperate，They always don't work as I expect them to. How can I use them in real projects?

First, Nginx is definitely the best choice！But...For small and medium-sized projects, most people still deploy the front end and back end together to simplify deployment.

1、Static
I want to put the front-end project I built with VueJS into the "./public" directory and then access it by visiting "http://127.0.0.1:8000". 
At the same time, I will assign the "/api" route to provide restful api for front-end projects

So I wrote the reference document like this:

public folder:
```
...
--public
---|css
----xxx.css
---|js
----xxx.js
---|img
----xxx.jpg
---|fonts
----xxx.woff
---index.html
--main.go
```

main.go:
```go
app.Static("/", "./public")
```
The actual result is: 
When I type "http://127.0.0.1:8000" in the browser, it will not automatically jump to "http://127.0.0.1:8000/" ,I need to manually modify the url to "http://127.0.0.1:8000/index.html" to see the content of "./public/index.html" in the browser, this is not over yet because index.html A lot of css, js, image, font files are referenced by relative paths. Now some of them can be downloaded and read correctly, and there are some path matching errors. My front-end application can't run normally, Static () method seems to be unable to distinguish The difference between directories and files...

2、Gzip

Compress () looks like a very nice feature. But in my experience, in most cases, it is a function for compressing static files. I just hope that there is a middleware that can be applied to the content that needs to be compressed. If it is smart enough, even No manual configuration is required. css, js, restful api can be turned on, image turned on but it is negative. > 4kb files can be opened, too small files are also negative. For the configuration, I just want it to be able to customize the compression level to reduce the CPU load.

If Compress is turned on globally, I don't have a way to turn it off where it is not needed, I can only turn off the entire routing group.

If turned off globally，I can't add it to static files because Static () is not a middleware. I dare not add it to other routing groups because it does not recognize how large the response body needs to be compressed.

Sorry for the above! I've used Fiber to build a project scaffolding, and I'm ready to use it for actual projects, because these features are very annoying.

Fiber is a great framework. It makes learning Go extremely fast. I hope it gets better and better!

**Comment by Fenny** [comment]
@hyingreborn, we moved compression to a middleware package: https://github.com/gofiber/compression, does this help for you? By default it gzips/deflates ( depending on request header ) above 4kb.

**Comment by hyingreborn** [comment]
@Fenny iconfont files still Unable to load properly

**Comment by Fenny** [comment]
This should be fixed in `v1.8.32` https://github.com/gofiber/fiber/releases/tag/v1.8.32
Feel free to re-open this issue if you still have problems regarding this issue.
